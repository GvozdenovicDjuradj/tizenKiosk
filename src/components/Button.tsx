import React from "react"
import {
  Text,
  TouchableOpacity,
  TouchableOpacityProps,
} from "react-native"

export interface ButtonProps extends TouchableOpacityProps {
  text?: string
  textColor?: string
  textSize?: number
  textFont?: string
}

export default (props: ButtonProps) => {
  const {
    text,
    textColor,
    textSize = 18,
    textFont = "Arial",
    ...rest
  } = props
  const textStyle = {
    color: textColor,
    fontSize: textSize,
    fontFamily: textFont
  }
  return (
    <TouchableOpacity {...rest}>
      <Text style={textStyle}>
        {text}
      </Text>
    </TouchableOpacity>
  )
}
