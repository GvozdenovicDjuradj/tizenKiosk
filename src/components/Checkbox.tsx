import React from "react"
import CheckBox, { CheckBoxProps } from "react-native-check-box"

export interface CheckboxProps {
  checkBoxColor?: CheckBoxProps["checkBoxColor"],
  checkedCheckBoxColor?: CheckBoxProps["checkedCheckBoxColor"],
  checkedImage?: CheckBoxProps["checkedImage"],
  disabled?: CheckBoxProps["disabled"],
  indeterminateImage?: CheckBoxProps["indeterminateImage"],
  isChecked?: CheckBoxProps["isChecked"],
  isIndeterminate?: CheckBoxProps["isIndeterminate"],
  leftText?: CheckBoxProps["leftText"],
  leftTextStyle?: CheckBoxProps["leftTextStyle"],
  leftTextView?: CheckBoxProps["leftTextView"],
  onClick?: (isChecked: boolean) => void,
  rightText?: CheckBoxProps["rightText"],
  rightTextStyle?: CheckBoxProps["rightTextStyle"],
  rightTextView?: CheckBoxProps["rightTextView"],
  style?: CheckBoxProps["style"],
  uncheckedCheckBoxColor?: CheckBoxProps["uncheckedCheckBoxColor"],
  unCheckedImage?: CheckBoxProps["unCheckedImage"],
}

const clickHandler = (checked?: CheckboxProps["isChecked"], onClick?: CheckboxProps["onClick"]) =>
  onClick && onClick(!checked)

export default (props: CheckboxProps) => {
  const { isChecked, onClick, ...rest } = props
  return (
    <CheckBox
      {...rest}
      isChecked={isChecked}
      onClick={() => clickHandler(isChecked, onClick)}
    />
  )
}
