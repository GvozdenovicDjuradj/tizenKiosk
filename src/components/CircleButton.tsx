import React from "react"
import {
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  TouchableOpacityProps,
} from "react-native"
import { NavigationEvents } from "react-navigation"

const circleSide =  260

const styles = StyleSheet.create({
  circleText: {
    fontSize: 30,
    textAlign: "center",
  },
  circleView: {
    alignItems: "center",
    borderRadius: circleSide / 2,
    elevation: 7,
    height: circleSide,
    justifyContent: "center",
    marginBottom: 10,
    padding: 5,
    shadowColor: "#000",
    shadowOffset: { width: 1, height: 1 },
    shadowOpacity: 0.8,
    shadowRadius: 10,
    width: circleSide,
  },
})

interface Props extends TouchableOpacityProps {
  text: string[];
  textStyle?: StyleProp<TextStyle>;
}

export default class extends React.Component<Props> {
  public state = {
    isDisabled: false
  }

  public disableHandler = (isDisabled: boolean) => () => {
    this.setState({
      isDisabled
    })
  }

  public render() {
    const { text, textStyle, style, ...otherProps } = this.props
    return (
      <React.Fragment>
        <NavigationEvents
          onWillFocus={this.disableHandler(false)}
          onWillBlur={this.disableHandler(true)}
        />
        <TouchableOpacity
          style={[styles.circleView, style]}
          disabled={this.state.isDisabled}
          {...otherProps}
        >
          {text.map((str: string, i: number) => {
            const txtStyle = Array.isArray(textStyle) ? textStyle[i] : textStyle
            return str ? (
              <Text key={i} style={[styles.circleText, txtStyle]}>{str}</Text>
            ) : null
          })}
        </TouchableOpacity>
      </React.Fragment>
    )
  }
}
