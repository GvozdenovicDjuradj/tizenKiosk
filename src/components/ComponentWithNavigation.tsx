import React from "react"
import { BackHandler } from "react-native"
import {
  NavigationActions,
  NavigationEventPayload,
  NavigationEventSubscription,
  NavigationScreenProp,
} from "react-navigation"

export interface ComponentWithNavigationProps {
  goBackExtra: () => any;
  navigation: NavigationScreenProp<any>;
}

export default class ComponentWithNavigation<T extends ComponentWithNavigationProps> extends React.PureComponent<T> {

  private wipeListener?: NavigationEventSubscription

  public componentWillMount() {
    const { navigation } = this.props
    BackHandler.addEventListener("hardwareBackPress", this.goBack)
    this.wipeListener = navigation.addListener("willBlur", this.gestureBackHandler)
  }

  public componentWillUnmount() {
    BackHandler.removeEventListener("hardwareBackPress", this.goBack)
    if (this.wipeListener) {
      this.wipeListener.remove()
    }
  }

  public gestureBackHandler = (payload: NavigationEventPayload) => {
    if (payload.action.type === NavigationActions.BACK) {
      if (this.props.goBackExtra) {
        this.props.goBackExtra()
      }
    }
  }

  public goBack = () => {
    this.props.navigation.dispatch(NavigationActions.back())
    return true
  }

}
