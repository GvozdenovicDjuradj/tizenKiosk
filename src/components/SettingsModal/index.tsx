import React from "react"
import {
  BackHandler,
  ScrollView,
  Switch,
  Text,
  TouchableOpacity,
  View,
} from "react-native"
import { connect } from "react-redux"
import DeviceInfo from "react-native-device-info"
import { DeviceInfoContainer } from "./deviceInfo"
import TextField from "../TextField"
import Button from "../Button"
import styles from "./styles"
import { FieldUpdatePayload, RegisterKioskFields, RootState } from "../../interfaces"
import { KioskSettingsForm } from "../../reducers/kioskSettings"
import {
  checkNetwork as checkNetworkStatus,
  deactivateApp,
  saveKioskSettingsChanges,
  updateSettingsField,
  validateKioskSettings,
  getPermissionReadPhoneState
} from "../../actions"
import { selectors, validators } from "../../utils"
import { colors } from "../../theme"
import {isPlatformAndroidOrIos} from "../../utils/Platform"
import PrinterConfiguration from "../../printer/containers/PrinterConfiguration"
import DebugOutput from "../../printer/containers/DebugOutput"

export interface SettingsModalProps {
  canSaveChanges: boolean
  canValidateSettings: boolean
  checkNetwork: () => void
  deactivate: () => void
  dismiss: () => void
  form: KioskSettingsForm
  saveChanges: () => void
  updateField: (payload: FieldUpdatePayload) => void
  validateSettings: () => void,
  deviceImei?: string | null,
  getPhoneStatePermission: () => void,
  isDebugOutputVisible: boolean,
}

interface SettingsModalState {
  deactivate: boolean
}

export class SettingsModal extends React.PureComponent<SettingsModalProps, SettingsModalState> {

  public state = { deactivate: false }

  public componentDidMount() {
    this.props.getPhoneStatePermission()
  }

  public render() {
    const {
      canSaveChanges,
      canValidateSettings,
      checkNetwork,
      deactivate,
      dismiss,
      form,
      updateField,
      validateSettings,
      deviceImei,
      isDebugOutputVisible
    } = this.props
    const deactivateBtnText = this.state.deactivate
      ? "Cancel deactivation"
      : "Deactivate Kiosk"
    return (
      <View style={styles.wrapper}>
        <View style={styles.header}>
          <Text style={styles.headerText}>
            Kiosk Settings
          </Text>
          <TouchableOpacity onPress={dismiss}>
            <Text style={styles.fa}>&#xf00d;</Text>
          </TouchableOpacity>
        </View>
        <ScrollView style={styles.contentWrapper}>
          <View>
            <View style={{ width: "100%", marginBottom: 10 }}>
              <View>
                <Text style={styles.textStyles}>URL</Text>
              </View>
              <View>
                <TextField
                  name="newKioskUrl"
                  onChangeText={(value: string) => updateField({
                    name: RegisterKioskFields.url,
                    value
                  })}
                  placeholder={"kiosk URL"}
                  style={styles.inputStyles}
                  validate={__DEV__
                    ? validators.strlen
                    : [validators.strlen, validators.qudiniUrl]
                  }
                  value={form.url}
                />
              </View>
            </View>
            <View style={{ width: "100%", marginBottom: 10 }}>
              <View>
                <Text style={styles.textStyles}>Kiosk ID</Text>
              </View>
              <View style={{ flexDirection: "row" }}>
                <TextField
                  name="kioskId"
                  onChangeText={(value: string) => updateField({
                    name: RegisterKioskFields.kioskIdentifier,
                    value
                  })}
                  placeholder={"kiosk ID"}
                  style={styles.inputStyles}
                  value={form.kioskIdentifier}
                />
              </View>
            </View>
            { isPlatformAndroidOrIos() &&
              <View style={[styles.row, { marginBottom: 10 }]}>
                <View>
                  <Text style={[styles.textStyles, { fontSize: 20 }]}>
                    Kiosk Mode
                  </Text>
                </View>
                <Switch
                    value={form.hasKioskModeEnable}
                    onValueChange={(value) => updateField({
                      name: RegisterKioskFields.hasKioskModeEnable,
                      value
                    })}
                />
              </View>
            }
          </View>
          { isPlatformAndroidOrIos() &&
            <View style={{ marginBottom: 10 }}>
              <PrinterConfiguration />
              { isDebugOutputVisible && <DebugOutput /> }
            </View>
          }
          <DeviceInfoContainer
              uniqueId={DeviceInfo.getUniqueID()}
              version={DeviceInfo.getReadableVersion()}
              deviceImei={deviceImei}
          />
          <View style={[styles.row, { marginBottom: 10 }]}>
            <Button
              disabled={!canValidateSettings}
              onPress={validateSettings}
              style={[styles.button, canValidateSettings ? {} : { opacity: 0.7 }]}
              text={"Validate Settings"}
              textColor={colors.text}
            />
            <Button
              text={"Check Network"}
              onPress={checkNetwork}
              style={styles.button}
              textColor={colors.text}
            />
            <Button
              text={deactivateBtnText}
              style={styles.button}
              onPress={this.toggleDeactivate}
              textColor={colors.text}
            />
          </View>
          {this.state.deactivate ? (
            <View style={[styles.row, styles.deactivateWarningView]}>
              <View style={{ flex: 1 }}>
                <Text style={{ fontSize: 14 }}>
                  Deactivating will clear this app so that it will need repairing
                </Text>
              </View>
              <TouchableOpacity onPress={deactivate} style={styles.deactivateBtn}>
                <Text style={styles.deactivateBtnText}>
                  Confirm deactivation
                </Text>
              </TouchableOpacity>
            </View>
          ) : null}
          <View style={{ flex: 1, marginBottom: 10 }}>
            <ScrollView style={styles.messagesView}>
              {form.diagnosticMessages.length
                ? form.diagnosticMessages.map((msg, idx) =>
                  <Text style={styles.textStyles} key={`diagnostic_${idx}`}>
                    {msg}
                  </Text>
                )
                : (
                  <Text style={{ opacity: 0.7 }}>
                    This is where network or validation messaging would appear
                  </Text>
                )
              }
            </ScrollView>
          </View>
          <View style={styles.bottomRow}>
            <Button
              disabled={!canSaveChanges}
              style={[styles.button, canSaveChanges ? {} : { opacity: 0.7 }]}
              text={"Save Settings"}
              textColor={colors.text}
            />
            <Button
              onPress={BackHandler.exitApp}
              style={styles.button}
              text={"Minimize app"}
              textColor={colors.text}
            />
          </View>
        </ScrollView>
      </View>
    )
  }

  private readonly toggleDeactivate = () =>
    this.setState({ deactivate: !this.state.deactivate })

}

const mapStateToProps = (state: RootState) => ({
  canSaveChanges: selectors.canSaveChanges(state),
  canValidateSettings: selectors.canValidateSettings(state),
  form: state.kioskSettings,
  deviceImei: state.app.deviceImei,
  isDebugOutputVisible: state.printer.isDebugOutputVisible
})

const mapDispatchToProps = {
  checkNetwork: checkNetworkStatus,
  deactivate: deactivateApp,
  saveChanges: saveKioskSettingsChanges,
  updateField: updateSettingsField,
  validateSettings: validateKioskSettings,
  getPhoneStatePermission: getPermissionReadPhoneState
}

export default connect(mapStateToProps, mapDispatchToProps)(SettingsModal)
