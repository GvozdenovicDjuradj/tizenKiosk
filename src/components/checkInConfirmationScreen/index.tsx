import React from "react"
import { Dispatch } from "redux"
import { connect } from "react-redux"
import { Image, ImageStyle, Text, View } from "react-native"
import Button from "../Button"
import Header from "../Header"
import TopBar from "../TopBar"
import { NavButtonBack } from "../NavButton"
import ComponentWithNavigation, { ComponentWithNavigationProps } from "../ComponentWithNavigation"
import { CheckInData, KioskTemplate, RootState } from "../../interfaces"
import { getTemplateBackground, getTemplateStyles, selectors, getLocaleString } from "../../utils"
import { Action } from "../../actions/types"
import { goToInitialScreen } from "../../actions"
import styles from "./styles"
import { mixins } from "../../theme"

interface PropsFromState {
  checkInData?: CheckInData;
  host: string;
  showQudiniLogo: boolean;
  template?: KioskTemplate;
}

interface PropsFromDispatch {
  goBack: () => any;
}

type Props = PropsFromState & PropsFromDispatch & ComponentWithNavigationProps

const getContent = (checkInData: CheckInData, textStyle: object) => (
  <View style={styles.row}>
    <View style={styles.grid}>
      <View style={styles.row}>
        <View style={styles.col}>
          <Text style={[styles.text, textStyle]}>
            {getLocaleString(`confirmationScreen.top.subText.name`) || "Name:"}
          </Text>
        </View>
        <View style={styles.col}>
          <Text style={[styles.text, textStyle]}>
            {checkInData.customer.name}
          </Text>
        </View>
      </View>
      <View style={styles.row}>
        <View style={styles.col}>
          <Text style={[styles.text, textStyle]}>
          {getLocaleString(`confirmationScreen.top.subText.bookingReference`) || "Booking ref:"}
          </Text>
        </View>
        <View style={styles.col}>
          <Text style={[styles.text, textStyle]}>
            {checkInData.customer.bookingRef}
          </Text>
        </View>
      </View>
    </View>
  </View>
)

class CheckInConfirmation extends ComponentWithNavigation<Props> {

  public goBack = () => {
    this.props.goBack()
    return true
  }

  public render() {
    const { checkInData, goBack, host, template } = this.props
    const templateBackground = getTemplateBackground(template, host)
    const templateStyles = getTemplateStyles(template, host)
    const button = { backgroundColor: "" }
    const textStyle = { color: "", fontFamily: "" }
    const buttonText = { color: "" }
    if (template) {
      button.backgroundColor = template.welcomeButtonColor
      buttonText.color = template.buttonTextColor
      textStyle.color = template.secondaryTextColor
      textStyle.fontFamily = template.font
    }
    const content = checkInData ? getContent(checkInData, textStyle) : null
    return (
      <View style={[styles.view, templateBackground.color]}>
        <TopBar>
          <View style={styles.navView}>
            <View style={{ flex: 1 }}>
              <NavButtonBack
                buttonColor={button.backgroundColor}
                onPress={goBack}
                text={getLocaleString(`confirmationScreen.navigation.back`) || "Home"}
                textColor={buttonText.color}
              />
            </View>
            {templateStyles.logo ?
              <Image
                resizeMode="contain"
                source={{ uri: templateStyles.logo }}
                style={styles.logo as ImageStyle}
              />
              : null}
            <View style={{ flex: 1 }}>
            </View>
          </View>
        </TopBar>
        <Header
          style={{ marginTop: 0 }}
          text={getLocaleString(`confirmationScreen.header.text.bookingCheckIn`) || "Thanks for checking in"}
          template={template}
        />
        <View style={styles.content}>
          {content}
        </View>
        <View style={[mixins.centerItem(true)]}>
          <Button
            onPress={goBack}
            style={[button, { padding: 15 }]}
            text={getLocaleString(`confirmationScreen.navigation.nextCustomer`) || "Next Customer"}
            textColor={buttonText.color}
            textSize={32}
            textFont={template && template.font}
          />
        </View>
        {templateBackground.image.url ? (
          <Image
            resizeMode="contain"
            style={styles.backgroundImage as ImageStyle}
            source={{ cache: "force-cache", uri: templateBackground.image.url }}
          />
        ) : null}
        {this.props.showQudiniLogo ? (
          <View style={styles.qudiniLogoView}>
            <Image
              source={require("../../../assets/images/logo-qudini.png")}
              style={styles.qudiniLogo as ImageStyle}
            />
          </View>
        ) : null}
      </View>
    )
  }
}

const mapStateToProps = (state: RootState): PropsFromState => ({
  checkInData: state.checkIn.data,
  host: state.kiosk.fields.url,
  showQudiniLogo: selectors.showQudiniLogo(state),
  template: state.kiosk.settings && state.kiosk.settings.template,
})

const mapDispatchToProps = (dispatch: Dispatch<Action>): PropsFromDispatch => ({
  goBack: () => dispatch(goToInitialScreen())
})

export default connect(mapStateToProps, mapDispatchToProps)(CheckInConfirmation)
