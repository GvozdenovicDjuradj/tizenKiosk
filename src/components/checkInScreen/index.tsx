import React from "react"
import { Dispatch } from "redux"
import { connect } from "react-redux"
import { Image, ImageStyle, View } from "react-native"
import Content from "./Content"
import Header from "../Header"
import TopBar from "../TopBar"
import { NavButtonBack, NavButtonNext } from "../NavButton"
import KeyboardResponsiveScrollView from "../keyboard/KeyboardResponsiveScrollView"
import ComponentWithNavigation, { ComponentWithNavigationProps } from "../ComponentWithNavigation"
import { checkIn } from "../../actions"
import { Action, KIOSK } from "../../actions/types"
import { AppScreens, KioskTemplate, RootState } from "../../interfaces"
import { getTemplateBackground, getTemplateStyles, selectors, getLocaleString } from "../../utils"
import { keyboardView as styles } from "./styles"

interface PropsFromState {
  host?: string;
  isInvalid: boolean;
  showQudiniLogo: boolean;
  template?: KioskTemplate;
}

interface PropsFromDispatch {
  goBackExtra: () => any;
  goNext: () => any;
}

type Props = PropsFromState & PropsFromDispatch & ComponentWithNavigationProps

class CheckInScreen extends ComponentWithNavigation<Props> {
  public render() {
    const { goNext, host, template } = this.props
    const templateBackground = getTemplateBackground(template, host)
    const templateStyles = getTemplateStyles(template, host)
    return (
      <KeyboardResponsiveScrollView
        contentContainerStyle={[styles.main, templateBackground.color]}
      >
        <TopBar>
          <View style={styles.navView}>
            <View style={{ flex: 1 }}>
              <NavButtonBack
                buttonColor={templateStyles.circle.backgroundColor}
                onPress={this.goBack}
                text={getLocaleString(`customerScreen.navigation.previous`) || "Back"}
                textColor={templateStyles.text.color}
              />
            </View>
            {templateStyles.logo ?
              <Image
                resizeMode="contain"
                source={{ uri: templateStyles.logo }}
                style={styles.logo as ImageStyle}
              />
              : null}
            <View style={{ flex: 1 }}>
              <NavButtonNext
                buttonColor={templateStyles.circle.backgroundColor}
                onPress={goNext}
                text={getLocaleString(`customerScreen.navigation.next`) || "Continue"}
                textColor={templateStyles.text.color}
              />
            </View>
          </View>
        </TopBar>
        <Header
          style={{ marginTop: 0 }}
          text={getLocaleString(`customerScreen.header.text.bookingCheckIn`) ||
          "Enter your details so we can find your booking"}
          template={template}
        />
        <Content />
        {this.props.showQudiniLogo ? (
          <View style={styles.bottomView}>
            <View style={{ flex: 1 }}>
            </View>
            <View style={styles.qudiniLogoView}>
              <Image
                source={require("../../../assets/images/logo-qudini.png")}
                style={styles.qudiniLogo as ImageStyle}
              />
            </View>
          </View>
        ) : null}
        {templateBackground.image.url ? (
          <Image
            resizeMode="contain"
            style={styles.backgroundImage as ImageStyle}
            source={{ cache: "force-cache", uri: templateBackground.image.url }}
          />
        ) : null}
      </KeyboardResponsiveScrollView>
    )
  }
}

const isInvalidSelector = (state: RootState) => {
  const nodata = (
    Boolean(state.checkIn.email) === false &&
    Boolean(state.checkIn.mobileNumber) === false &&
    Boolean(state.checkIn.orderNumber) === false
  )
  if (nodata) {
    return true
  }
  return !selectors.isScreenValid(AppScreens.CHECK_IN)(state)
}

const mapStateToProps = (state: RootState): PropsFromState => ({
  host: state.kiosk.fields.url,
  isInvalid: isInvalidSelector(state),
  showQudiniLogo: selectors.showQudiniLogo(state),
  template: state.kiosk.settings && state.kiosk.settings.template,
})

const mapDispatchToProps = (dispatch: Dispatch<Action>): PropsFromDispatch => ({
  goBackExtra: () => dispatch({ type: KIOSK.CHECK_IN.RESET }),
  goNext: () => dispatch(checkIn())
})

export default connect(mapStateToProps, mapDispatchToProps)(CheckInScreen)
