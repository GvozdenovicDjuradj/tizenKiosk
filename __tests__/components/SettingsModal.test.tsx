import React from "react"
import ShallowRenderer from "react-test-renderer/shallow";
import { SettingsModal, SettingsModalProps } from "../../src/components/SettingsModal"

describe("SettingsModal component test", () => {
    it("should render SettingsModal successfully", () => {
        // Given
        const renderer = new ShallowRenderer();
        const props: SettingsModalProps = {
            canSaveChanges: true,
            canValidateSettings: true,
            checkNetwork: () => {},
            deactivate: () => {},
            dismiss: () => {},
            form: {
                diagnosticMessages: [],
                kioskIdentifier: "",
                printerUrl: "",
                hasPrinter: true,
                hasKioskModeEnable: true,
                url: "",
            },
            saveChanges: () => {},
            updateField: () => {},
            validateSettings: () => {},
            deviceImei: "IMEI",
            getPhoneStatePermission: () => {},
        }

        // When
        renderer.render(
            <SettingsModal {...props} />
        )

        // Then
        expect(renderer.getRenderOutput()).toMatchSnapshot()
    })
})
