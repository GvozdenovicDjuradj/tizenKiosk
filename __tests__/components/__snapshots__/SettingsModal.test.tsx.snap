// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsModal component test should render SettingsModal successfully 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#000",
      "borderWidth": 1,
      "flex": 1,
      "height": "100%",
      "paddingBottom": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 0.1,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 26,
          "fontWeight": "bold",
        }
      }
    >
      Kiosk Settings
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "FontAwesome5FreeSolid",
            "fontSize": 32,
          }
        }
      >
        ÔÄç
      </Text>
    </TouchableOpacity>
  </View>
  <ScrollViewMock
    style={
      Object {
        "backgroundColor": "#fff",
        "flexGrow": 0.9,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 30,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginBottom": 10,
            "width": "100%",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#000",
              }
            }
          >
            URL
          </Text>
        </View>
        <View>
          <Connect(WrappedInput)
            name="newKioskUrl"
            onChangeText={[Function]}
            placeholder="kiosk URL"
            style={
              Object {
                "color": "#000",
                "fontSize": 16,
                "height": 38,
                "lineHeight": 18,
              }
            }
            validate={[Function]}
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
            "width": "100%",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#000",
              }
            }
          >
            Kiosk ID
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Connect(WrappedInput)
            name="kioskId"
            onChangeText={[Function]}
            placeholder="kiosk ID"
            style={
              Object {
                "color": "#000",
                "fontSize": 16,
                "height": 38,
                "lineHeight": 18,
              }
            }
            value=""
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "minHeight": 50,
            },
            Object {
              "marginBottom": 10,
            },
          ]
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "fontSize": 20,
                },
              ]
            }
          >
            Kiosk Mode
          </Text>
        </View>
        <Switch
          onValueChange={[Function]}
          value={true}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Connect(_default) />
    </View>
    <DeviceInfoContainer
      deviceImei="IMEI"
      uniqueId="1234567890"
      version="1.0.1"
    />
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
          },
          Object {
            "marginBottom": 10,
          },
        ]
      }
    >
      <_default
        disabled={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#dfdfdf",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "paddingHorizontal": 10,
            },
            Object {},
          ]
        }
        text="Validate Settings"
        textColor="#000"
      />
      <_default
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#dfdfdf",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "paddingHorizontal": 10,
          }
        }
        text="Check Network"
        textColor="#000"
      />
      <_default
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#dfdfdf",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "paddingHorizontal": 10,
          }
        }
        text="Deactivate Kiosk"
        textColor="#000"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginBottom": 10,
        }
      }
    >
      <ScrollViewMock
        style={
          Object {
            "borderColor": "#000",
            "borderWidth": 1,
            "padding": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "opacity": 0.7,
            }
          }
        >
          This is where network or validation messaging would appear
        </Text>
      </ScrollViewMock>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 120,
          "justifyContent": "space-between",
        }
      }
    >
      <_default
        disabled={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#dfdfdf",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "paddingHorizontal": 10,
            },
            Object {},
          ]
        }
        text="Save Settings"
        textColor="#000"
      />
      <_default
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#dfdfdf",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "paddingHorizontal": 10,
          }
        }
        text="Minimize app"
        textColor="#000"
      />
    </View>
  </ScrollViewMock>
</View>
`;
